packages/vuetify/src/index.ts

```diff
@@ -1,19 +1,58 @@
+// Core
+import Vuetify from './framework'
+import { install } from './install'
+
+// Functionality
 import * as components from './components'
 import * as directives from './directives'
-import Vuetify from './framework'
 
-export default Vuetify
+// Vue
+import {
+  inject,
+  provide,
+} from 'vue'
 
-const install = Vuetify.install
+const VuetifySymbol = Symbol.for('vuetify')
 
-Vuetify.install = (Vue, args) => {
-  install.call(Vuetify, Vue, {
+// Ordering for plugin install invocation was changed and won't work for us atm
+// Hoping to remove this in the future or rename it
+// v2.x https://github.com/vuejs/vue/blob/6fe07ebf5ab3fea1860c59fe7cdd2ec1b760f9b0/src/core/global-api/use.js#L15
+// v3.x https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/src/apiCreateApp.ts#L106
+function VuetifyInstall (app: any, args: any) {
+  install(app, {
     components,
     directives,
     ...args,
   })
 }
 
-if (typeof window !== 'undefined' && window.Vue) {
-  window.Vue.use(Vuetify)
+function createVuetify (opts: any) {
+  const vuetify = new Vuetify(opts)
+
+  vuetify.init()
+
+  provide(VuetifySymbol, vuetify.framework)
 }
+
+function useVuetify () {
+  const vuetify = inject(VuetifySymbol)
+
+  if (!vuetify) {
+    throw Error(`Unable to find vuetify instance on Symbol <${String(VuetifySymbol)}>`)
```

Comment:
This will most likely be a strange message for end-users. I think something that mentions that `createVuetify` has not been called would be better.